homeassistant/components/shelly/utils.py

```diff
@@ -220,12 +222,22 @@ def get_shbtn_input_triggers() -> list[tuple[str, str]]:
 async def get_coap_context(hass: HomeAssistant) -> COAP:
     """Get CoAP context to be used in all Shelly Gen1 devices."""
     context = COAP()
+
+    adapters = await network.async_get_adapters(hass)
+    LOGGER.debug("Network adapters: %s", adapters)
+    if network.async_only_default_interface_enabled(adapters):
+        ipv4 = DEFAULT_IP_ADDRESS
+    else:
+        for adapter in adapters:
+            if adapter["enabled"] and adapter["default"]:
```

Comment:
Should this be handling multiple adapters instead of picking the default?

Reply:
Yes, I think we can always pass a list to the socket initialize method. Will test it

Reply:
I think we should try the default one first, than add the others so we can catch `OSError` on subsequent ones so we don't have a repeat of https://github.com/home-assistant/core/issues/45957

Reply:
I followed https://github.com/home-assistant/architecture/discussions/509#discussioncomment-672957
Should be safe.